### PR TITLE
ENH: more bti tweaks

### PR DESCRIPTION
**Pull request by dengemann**
cc @Eric89GXL @agramfort

### CODE REVIEW

**Comment by agramfort** [comment]
cannot tell if it's good. I trust you. Merge if you're happy

**Comment by dengemann** [comment]
> cannot tell if it's good. I trust you. Merge if you're happy

Not done ... more to come.
But I think you can. The goal is to get more control over steps and seperate processing and convertig for testing and other purposes.

**Comment by dengemann** [comment]
@agramfort ready for review. Test pass locally. Implemented a clean + tested no-convert options.

File: mne/io/bti/bti.py [hunk 1]
```diff
@@ -179,30 +171,54 @@ def _setup_head_shape(fname, use_hpi=True):
...
-    idx_points, dig_points, t = _convert_head_shape(idx_points, dig_points)
-    all_points = np.r_[idx_points, dig_points].astype('>f4')
+    if convert:
+        ctf_head_t = _get_ctf_head_to_head_t(idx_points)
+    else:
+        ctf_head_t = Transform('ctf_head', 'ctf_head', np.eye(4))
+
+    if dig_points is not None:
+        # dig_points = apply_trans(ctf_head_t['trans'], dig_points)
+        all_points = np.r_[idx_points, dig_points]
```
Comment (Eric89GXL) [inline mne/io/bti/bti.py]: np.concatenate is preferred to r_ and c_

File: mne/io/bti/bti.py [hunk 2]
```diff
@@ -1012,13 +1035,15 @@ def __init__(self, pdf_fname, config_fname='config',
...
                               _correct_trans(bti_info['bti_transform'][0]))
         # for old backward compatibility
         rotation_x = 0. if rotation_x is None else rotation_x
-        bti_dev_t = Transform('ctf_meg', 'meg',
-                              _get_bti_dev_t(rotation_x, translation))
+        if convert:
+            bti_dev_t = Transform('ctf_meg', 'meg',
+                                  _get_bti_dev_t(rotation_x, translation))
+        else:
+            bti_dev_t = np.eye(4)
```
Comment (Eric89GXL) [inline mne/io/bti/bti.py]: better as:
```
if convert:
    bti_dev_t = _get_bti_dev_t(rotation_x, translation)
else:
    bti_dev_t = np.eye(4)
bti_dev_t = Transform('ctf_meg', 'meg', bti_dev_t)
```

File: mne/io/bti/bti.py [hunk 3]
```diff
@@ -1209,10 +1241,13 @@ def read_raw_bti(pdf_fname, config_fname='config',
...
     rotation_x : float
-        Degrees to tilt x-axis for sensor frame misalignment.
+        Degrees to tilt x-axis for sensor frame misalignment. Ignored
+        if convert is True.
     translation : array-like, shape (3,)
         The translation to place the origin of coordinate system
-        to the center of the head.
+        to the center of the head. Ignored if convert is True.
+    convert : bool
+        Convert to Neuromag or not.
```
Comment (Eric89GXL) [inline mne/io/bti/bti.py]: Neuromag coordinates

File: mne/io/bti/bti.py [hunk 4]
```diff
@@ -141,25 +141,17 @@ def _convert_head_shape(idx_points, dig_points):
...
-    if dig_points is not None:
-        dig_points_nm = apply_trans(t['trans'], dig_points)
-    else:
-        dig_points_nm = None
-    return idx_points_nm, dig_points_nm, t
+    return idx_points_nm
 
 
-def _setup_head_shape(fname, use_hpi=True):
+def _processes_bti_headshape(fname, convert=True, use_hpi=True):
```
Comment (Eric89GXL) [inline mne/io/bti/bti.py]: process instead of processes

File: mne/io/bti/tests/test_bti.py [hunk 5]
```diff
@@ -79,59 +103,95 @@ def test_raw():
...
+            assert_true(isinstance(re.info[key], dict))
+            this_t = re.info[key]['trans']
+            assert_equal(this_t.shape, (4, 4))
+            # cehck that matrix by is not identity
+            assert_true(not np.allclose(this_t, np.eye(4)))
         os.remove(tmp_raw_fname)
 
 
+def test_no_conversion():
+    """ Test bti conversion to Raw object """
```
Comment (Eric89GXL) [inline mne/io/bti/tests/test_bti.py]: should have different docstring from other testing function

**Comment by Eric89GXL** [comment]
Travis is unhappy

**Comment by dengemann** [comment]
Ok Trvavis should soon be happy. I'm off for a moment.

**Comment by Eric89GXL** [comment]
LGTM, +1 for merge

**Comment by agramfort** [comment]
LGTM

merge if you're happy as travis is :)

**Pull request merged**